Sources/FoundationEssentials/Calendar/Calendar_Recurrence.swift

```diff
@@ -0,0 +1,835 @@
+                // https://github.com/apple/swift-foundation/issues/483
+                // Calendar.dates(byMatching: ...) does not respect repeatedTimePolicy when .nanosecond is set #483
+                let components: Calendar.ComponentSet = switch recurrence.frequency {
+                    case .minutely: [.nanosecond, .second]
+                    case .hourly:   [.nanosecond, .second, .minute]
+                    case .daily:    [.nanosecond, .second, .minute, .hour]
+                    case .weekly:   [.nanosecond, .second, .minute, .hour, .weekday]
+                    case .monthly:  [.nanosecond, .second, .minute, .hour, .day]
+                    case .yearly:   [.nanosecond, .second, .minute, .hour, .day, .month]
+                }
+#else
+                let components: Calendar.ComponentSet = switch recurrence.frequency {
+                    case .minutely: [.second]
+                    case .hourly:   [.second, .minute]
+                    case .daily:    [.second, .minute, .hour]
+                    case .weekly:   [.second, .minute, .hour, .weekday]
+                    case .monthly:  [.second, .minute, .hour, .day]
+                    case .yearly:   [.second, .minute, .hour, .day, .month]
+                }
+#endif
+                let componentsForEnumerating = recurrence.calendar._dateComponents(components, from: start) 
+                
+                let rangeForBaseRecurrence: Range<Date>? = nil
+                baseRecurrence = Calendar.DatesByMatching(calendar: recurrence.calendar,
+                                                          start: start,
+                                                          range: rangeForBaseRecurrence,
+                                                          matchingComponents: componentsForEnumerating,
+                                                          matchingPolicy: recurrence.matchingPolicy,
+                                                          repeatedTimePolicy: recurrence.repeatedTimePolicy,
+                                                          direction: .forward).makeIterator()
+                                                          
+                searchLimit = switch recurrence.frequency {
+                    // We want to find a search limit that will stop an infinite
+                    // loop, but that also enumerates through results that users
+                    // might be looking for. Assume that the recurrence requests
+                    // that a date falls on a leap day that occurs on a specific
+                    // weekday, that would yield a result at least once every 40
+                    // years.
+                    case .yearly:   40
+                    case .monthly:  40 * 12
+                    case .weekly:   40 * 53
+                    case .daily:    40 * 366
+                    case .hourly:   40 * 366 * 24
+                    case .minutely: 40 * 366 * 24 * 60
```

Comment:
These limits are somewhat arbitrary, but ultimately we wouldn't want `RecurrenceRule` to be used as a predicate for finding rare dates